File: test/veryeasyswift/ArraySumTest.swift
import Testing

@testable import very_easy

@Test(
  "Array summation",
  arguments: [
    ([1_000_000_001, 1_000_000_002, 1_000_000_003, 1_000_000_004, 1_000_000_005], 5_000_000_015)
  ])
func arraySumTest(_ param: [Int], _ expected: Int) {
  #expect(arraySum(array: param) == expected)
}

File: very_easy/array_sum_test.go
package very_easy

import (
	"testing"
)

func TestArraySum(t *testing.T) {
	cases := []struct {
		array    []int
		expected int
	}{
		{[]int{12, 58, 7, 42, 79}, 198},
	}
	for _, c := range cases {
		t.Run("", func(t *testing.T) {
			result := ArraySum(c.array)
			if result != c.expected {
				t.Errorf("ArraySum(%d) == %v, expected %v", c.array, result, c.expected)
			}
		})
	}
}
